src/mainloop.cpp

```diff
@@ -390,6 +390,19 @@ bool Mainloop::add_endpoints(const Configuration &config)
         g_endpoints.emplace_back(tcp);
     }
 
+    // Link grouped endpoints together
+    for (auto e : g_endpoints) {
+        if (e->get_group_name().empty()) {
+            continue;
+        }
+
+        for (auto other : g_endpoints) { // find other endpoints in group
+            if (other != e && e->get_group_name() == e->get_group_name()) {
+                e->link_group_member(other);
+            }
+        }
+    }
```

Comment:
I am seeing the setup of a group, but I'm missing where you change the route rules to avoid sending the message to endpoints if they belong to the same group. Am I missing something?